Sources/Datadog/Core/System/Time/DateProvider.swift

```diff
@@ -1,18 +0,0 @@
-/*
- * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
- * This product includes software developed at Datadog (https://www.datadoghq.com/).
- * Copyright 2019-2020 Datadog, Inc.
- */
-
-import Foundation
-
-/// Provides current device time information.
-internal protocol DateProvider {
-    /// Current device time.
-    func currentDate() -> Date
-}
```

Comment:
This was a leftover after #913, breaking MacCatalyst compatibility (duplicated file).